e2e/cypress/integration/messaging/message_channel_reference_spec.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M18707 - Autocomplete should close if tildes are deleted using backspace', () => {
+        const msg = 'foo';
+
+        // # Make a post
+        cy.postMessage(msg);
+
+        // # Hit the "up" arrow to open the edit modal
+        cy.get('#post_textbox').type('{uparrow}');
+
+        // # Insert a tilde (~) at the beginning of the post to be edited
+        cy.get('#edit_textbox').should('be.visible').wait(TIMEOUTS.TINY).type('{ctrl}{home}').type('~');
```

Comment:
I wonder why there's `{ctrl}`.  I run without it and works fine for me.  Could it be platform specific?  I'm on Mac by the way.

Same case in [line 34](https://github.com/mattermost/mattermost-webapp/pull/3999/files#diff-37684c12772a269c7c76457cb5a950dfR34)

Reply:
`Ctrl + Home` means "go to the beginning of the document" while `Home` means "go to the beginning of the **current line**" (I referred to https://www.w3schools.com/tags/ref_keyboardshortcuts.asp).

In this case both of them is OK, but I thought the former is better because it works fine even for multi-line message.
I can remove `{ctrl}` if you'd like.

Reply:
Oh interesting.  I'm good as is then.